pattern.go

```diff
@@ -196,12 +197,33 @@ func readSpecialPattern(pb *patternBuild, valsIn []typedVal) (pathVals []typedVa
 		pathVals, err = readExistsSpecial(pb, pathVals)
 	case "shellstyle":
 		pathVals, err = readShellStyleSpecial(pb, pathVals)
+	case "prefix":
+		pathVals, err = readPrefixSpecial(pb, pathVals)
 	default:
 		err = errors.New("unrecognized in special pattern: " + tt)
 	}
 	return
 }
 
+func readPrefixSpecial(pb *patternBuild, valsIn []typedVal) (pathVals []typedVal, err error) {
+	t, err := pb.jd.Token()
+	if err != nil {
+		return
+	}
+	pathVals = valsIn
+
+	prefixString, ok := t.(string)
+	if !ok {
+		err = errors.New("value for 'prefix' must be a string")
+		return
+	}
+	pathVals = append(pathVals, typedVal{vType: prefixType, val: `"` + prefixString + `"`})
```

Comment:
nit: improve readability by assigning `typedVal{vType: prefixType, val: `"` + prefixString + `"`}` a variable and using the var in `append`

Reply:
This is what I meant and I feel it improves readability, but this is nit:

```go
val := typedVal{
	vType: prefixType,
	val:   fmt.Sprintf("%q", prefixString), // or keep the concat if you're concerned about allocs
}
pathVals = append(pathVals, val)
```

Reply:
OK, you're right, it adds readability.